packages/task/src/common/task-watcher.ts

```diff
@@ -16,31 +16,39 @@
 
 import { injectable } from 'inversify';
 import { Emitter, Event } from '@theia/core/lib/common/event';
-import { TaskClient, TaskExitedEvent, TaskInfo } from './task-protocol';
+import { TaskClient, TaskExitedEvent, TaskInfo, TaskOutputProcessedEvent } from './task-protocol';
 
 @injectable()
 export class TaskWatcher {
 
     getTaskClient(): TaskClient {
         const newTaskEmitter = this.onTaskCreatedEmitter;
         const exitEmitter = this.onTaskExitEmitter;
+        const outputProcessedEmitter = this.onOutputProcessedEmitter;
         return {
             onTaskCreated(event: TaskInfo) {
                 newTaskEmitter.fire(event);
             },
             onTaskExit(event: TaskExitedEvent) {
                 exitEmitter.fire(event);
+            },
+            onTaskOutputProcessed(event: TaskOutputProcessedEvent) {
```

Comment:
`onDidProcessTaskOutput` will be better, we have conventions to event naming similar to VS Code please check, somehow everybody comes with own naming anyway